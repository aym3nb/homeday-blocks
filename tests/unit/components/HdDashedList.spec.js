import { shallowMount } from '@vue/test-utils';
import HdDashedList from '@/components/HdDashedList.vue';
import ITEMS from '@/stories/mocks/DASHED_LIST_ITEMS';

const ITEM_SELECTOR = '.dashed-list__item';
const LIST_WITH_EXPANDED_LABELS_CLASS = 'dashed-list--expanded-label';

describe('HdDashedList', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(HdDashedList, {
      propsData: {
        items: ITEMS,
      },
    });
  });

  test('renders component', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders items correctly', () => {
    expect(wrapper.findAll(ITEM_SELECTOR).length).toEqual(ITEMS.length);
  });

  test('renders expanded labels if if the prop `expandedLabel` is true', () => {
    expect(wrapper.classes()).not.toContain(LIST_WITH_EXPANDED_LABELS_CLASS);

    wrapper.setProps({
      expandedLabel: true,
    });

    expect(wrapper.classes()).toContain(LIST_WITH_EXPANDED_LABELS_CLASS);
  });
});
