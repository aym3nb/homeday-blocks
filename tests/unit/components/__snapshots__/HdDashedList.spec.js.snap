// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HdDashedList renders component 1`] = `
<dl class="dashed-list">
  <!---->
  <div class="dashed-list__item">
    <dt class="dashed-list__section dashed-list__section--label"><span class="dashed-list__text">item</span>
      <div class="dashed-list__dashes"></div>
    </dt>
    <dd class="dashed-list__section"><span class="dashed-list__text">1</span></dd>
    <!---->
  </div>
  <!---->
  <div class="dashed-list__item">
    <dt class="dashed-list__section dashed-list__section--label"><span class="dashed-list__text">item</span>
      <div class="dashed-list__dashes"></div>
    </dt>
    <dd class="dashed-list__section"><span class="dashed-list__text">2</span></dd>
    <!---->
  </div>
  <!---->
  <div class="dashed-list__item">
    <dt class="dashed-list__section dashed-list__section--label"><span class="dashed-list__text">item</span>
      <div class="dashed-list__dashes"></div>
    </dt>
    <dd class="dashed-list__section"><span class="dashed-list__text">3</span></dd>
    <!---->
  </div>
  <!---->
  <div class="dashed-list__item">
    <dt class="dashed-list__section dashed-list__section--label"><span class="dashed-list__text">item</span>
      <div class="dashed-list__dashes"></div>
    </dt>
    <dd class="dashed-list__section"><span class="dashed-list__text">4</span></dd>
    <!---->
  </div>
  <!---->
  <div class="dashed-list__item">
    <dt class="dashed-list__section dashed-list__section--label"><span class="dashed-list__text">item</span>
      <div class="dashed-list__dashes"></div>
    </dt>
    <dd class="dashed-list__section"><span class="dashed-list__text">5</span></dd>
    <!---->
  </div>
</dl>
`;
